src/renderer/logic/io/FileIO.ts

```diff
@@ -0,0 +1,106 @@
+import { PathLike } from "fs";
+import { Playlist } from "../playlist";
+import { ALLOWED_AUDIO_EXTENSIONS, ALLOWED_EXTENSIONS, ALLOWED_PLAYLIST_FORMATS, AMETHYST_PLAYLIST_EXTENSION } from "@shared/constants";
+import { amethyst } from "@/amethyst";
+import fs from "fs";
+
+/** A simple class that contains functions for paths, file input and output */
+export abstract class FileIO {
+
+    /**
+     * This function stores a playlist to a file
+     * @param playlist the playlist that should be stored
+     * @param path the path where it should be stored at, if invalid, it will use the default path
+     * @param overwrite if true, it will overwrite the file with the same name, if present
+     * @returns if successful true, false otherwise
+     */
+    public static writePlaylist(playlist: Playlist, path: PathLike, overwrite: boolean): boolean { // TODO:
+        return false;
+    }
+
+    /**
+     * Returns a playlist stored as a file, or undefined if invalid
+     * @param path the path where the a playlist is located
+     * @returns a playlist instance if successful, undefined otherwise
+     */
+    public static readPlaylist(path: PathLike): Playlist | undefined { // TODO:
+        return undefined;
+    }
+
+    public static async writeFile(path: PathLike, data: String) {
+        fs.promises.writeFile(path, data).catch(error => console.log(error));
```

Comment:
this should work for every occurance of fs

```diff
- fs.promises.writeFile(path, data).catch(error => console.log(error));
+ window.fs.promises.writeFile(path, data).catch(error => console.log(error));
```

Reply:
Thank you, that works (without the .promises, since apparently window.fs is already fs.promises), but with that in the same class the function redirect the (await window.fs.stat(path)).isDirectory() call is now an error in the console while running, it says .isDirectory() it's not a function, what am I not understanding here?

Reply:
Nevermind, await window.fs.stat(path) returns an object-type for some reason, makes sense why it does not recognize the function, but how do I know if a path is a directory or not, that is the last thing I need to complete the drag and drop feature

Reply:
I got around it by trying to open it as if it were a directory and if it fails it proceeds as if it were a file, not the best either since I can't check if it exists but whatever, works in most cases